File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Resourcemanager/AlibabaCloud/SDK/ROS/CDK/Resourcemanager/IMessageContact.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Resourcemanager
{
    /// <summary>Represents a `MessageContact`.</summary>
    [JsiiInterface(nativeType: typeof(IMessageContact), fullyQualifiedName: "@alicloud/ros-cdk-resourcemanager.IMessageContact")]
    public interface IMessageContact : AlibabaCloud.SDK.ROS.CDK.Core.IResource
    {
        /// <summary>Attribute CreateTime: Creation time of Message Contact.</summary>
        [JsiiProperty(name: "attrCreateTime", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrCreateTime
        {
            get;
        }

        /// <summary>Attribute EmailAddress: The email address of the contact.</summary>
        [JsiiProperty(name: "attrEmailAddress", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrEmailAddress
        {
            get;
        }

        /// <summary>Attribute MessageContactId: The ID of Message Contact.</summary>
        [JsiiProperty(name: "attrMessageContactId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrMessageContactId
        {
            get;
        }

        /// <summary>Attribute MessageContactName: The name of the contact.</summary>
        [JsiiProperty(name: "attrMessageContactName", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrMessageContactName
        {
            get;
        }

        /// <summary>Attribute MessageTypes: The types of messages received by the contact.</summary>
        [JsiiProperty(name: "attrMessageTypes", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrMessageTypes
        {
            get;
        }

        /// <summary>Attribute PhoneNumber: The mobile phone number of the contact.</summary>
        [JsiiProperty(name: "attrPhoneNumber", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrPhoneNumber
        {
            get;
        }

        /// <summary>Attribute Title: The job title of the contact.</summary>
        [JsiiProperty(name: "attrTitle", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrTitle
        {
            get;
        }

        [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-resourcemanager.MessageContactProps\"}")]
        AlibabaCloud.SDK.ROS.CDK.Resourcemanager.IMessageContactProps Props
        {
            get;
        }

        /// <summary>Represents a `MessageContact`.</summary>
        [JsiiTypeProxy(nativeType: typeof(IMessageContact), fullyQualifiedName: "@alicloud/ros-cdk-resourcemanager.IMessageContact")]
        internal sealed class _Proxy : DeputyBase, AlibabaCloud.SDK.ROS.CDK.Resourcemanager.IMessageContact
        {
            private _Proxy(ByRefValue reference): base(reference)
            {
            }

            /// <summary>Attribute CreateTime: Creation time of Message Contact.</summary>
            [JsiiProperty(name: "attrCreateTime", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrCreateTime
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute EmailAddress: The email address of the contact.</summary>
            [JsiiProperty(name: "attrEmailAddress", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrEmailAddress
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute MessageContactId: The ID of Message Contact.</summary>
            [JsiiProperty(name: "attrMessageContactId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrMessageContactId
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute MessageContactName: The name of the contact.</summary>
            [JsiiProperty(name: "attrMessageContactName", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrMessageContactName
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute MessageTypes: The types of messages received by the contact.</summary>
            [JsiiProperty(name: "attrMessageTypes", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrMessageTypes
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute PhoneNumber: The mobile phone number of the contact.</summary>
            [JsiiProperty(name: "attrPhoneNumber", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrPhoneNumber
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute Title: The job title of the contact.</summary>
            [JsiiProperty(name: "attrTitle", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrTitle
            {
                get => GetInstanceProperty<object>()!;
            }

            [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-resourcemanager.MessageContactProps\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Resourcemanager.IMessageContactProps Props
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Resourcemanager.IMessageContactProps>()!;
            }

            /// <summary>The environment this resource belongs to.</summary>
            /// <remarks>
            /// For resources that are created and managed by the CDK
            /// (generally, those created by creating new class instances like Role, Bucket, etc.),
            /// this is always the same as the environment of the stack they belong to;
            /// however, for imported resources
            /// (those obtained from static methods like fromRoleArn, fromBucketName, etc.),
            /// that might be different than the stack they were imported into.
            /// </remarks>
            [JsiiProperty(name: "env", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.IResourceEnvironment\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment Env
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment>()!;
            }

            /// <summary>The stack in which this resource is defined.</summary>
            [JsiiProperty(name: "stack", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.Stack\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.Stack Stack
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.Stack>()!;
            }

            /// <summary>The construct tree node for this construct.</summary>
            [JsiiProperty(name: "node", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ConstructNode\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode Node
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode>()!;
            }
        }
    }
}

File: multiple-languages/go/alicloudroscdkresourcemanager/IMessageContact.go
package alicloudroscdkresourcemanager

import (
	_jsii_ "github.com/aws/jsii-runtime-go/runtime"

	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkcore"
	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkresourcemanager/internal"
)

// Represents a `MessageContact`.
type IMessageContact interface {
	alicloudroscdkcore.IResource
	// Attribute CreateTime: Creation time of Message Contact.
	AttrCreateTime() interface{}
	// Attribute EmailAddress: The email address of the contact.
	AttrEmailAddress() interface{}
	// Attribute MessageContactId: The ID of Message Contact.
	AttrMessageContactId() interface{}
	// Attribute MessageContactName: The name of the contact.
	AttrMessageContactName() interface{}
	// Attribute MessageTypes: The types of messages received by the contact.
	AttrMessageTypes() interface{}
	// Attribute PhoneNumber: The mobile phone number of the contact.
	AttrPhoneNumber() interface{}
	// Attribute Title: The job title of the contact.
	AttrTitle() interface{}
	Props() *MessageContactProps
}

// The jsii proxy for IMessageContact
type jsiiProxy_IMessageContact struct {
	internal.Type__alicloudroscdkcoreIResource
}

func (j *jsiiProxy_IMessageContact) AttrCreateTime() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrCreateTime",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IMessageContact) AttrEmailAddress() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrEmailAddress",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IMessageContact) AttrMessageContactId() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrMessageContactId",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IMessageContact) AttrMessageContactName() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrMessageContactName",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IMessageContact) AttrMessageTypes() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrMessageTypes",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IMessageContact) AttrPhoneNumber() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrPhoneNumber",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IMessageContact) AttrTitle() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrTitle",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IMessageContact) Props() *MessageContactProps {
	var returns *MessageContactProps
	_jsii_.Get(
		j,
		"props",
		&returns,
	)
	return returns
}

